app/pages/events/event_conversation.php

```diff
@@ -0,0 +1,70 @@
+<?php
+restrict_access();
+$event_id = get_route_param('event_id');
+$event = em()->find(Event::class, $event_id);
+$me_user = User::getMain();
+
+$v = new Validator(action: "new_message");
+$input = $v->text("new_message")->placeholder("Message...")->autocomplete("off");
+
+if ($v->valid()) {
+    $event->conversation->sendMessage($me_user, $input->value);
+}
+
+$v_delete = new Validator(action: "delete_message");
+if ($v_delete->valid() && isset($_POST['message_id'])) {
+    $message = em()->find(Message::class, $_POST['message_id']);
+    if ($message && ($message->sender == $me_user || check_auth(Access::$EDIT_USERS))) {
+        em()->remove($message);
+        em()->flush();
+    }
+}
+
+$conversation = isset($event->conversation) ? $event->conversation : null;
+if (!$conversation) {
+    $conversation = new Conversation();
+    $event->conversation = $conversation;
+    em()->persist($conversation);
```

Comment:
I believe you can have cascade persistence here. I might be wrong though

Reply:
Right